raiden/network/transport/matrix/transport.py

```diff
@@ -1012,12 +1015,46 @@ def _get_room_for_address(
             self.log.error('No valid peer found', peer_address=address_hex)
             return None
 
-        self._address_to_userids[address].update({user.user_id for user in peers})
-
         if self._private_rooms:
             room = self._get_private_room(invitees=peers)
         else:
             room = self._get_public_room(room_name, invitees=peers)
+
+        peer_ids = self._address_to_userids[address]
+        member_ids = {member.user_id for member in room.get_joined_members(force_resync=True)}
+        room_is_empty = not bool(peer_ids & member_ids)
+        if room_is_empty:
+            last_ex: Optional[Exception] = False
+            retry_interval = 0.1
+            self.log.debug(
+                'Waiting for peer to join from invite',
+                peer_address=address_hex,
+            )
+            for _ in range(JOIN_RETRIES):
+                try:
+                    member_ids = {member.user_id for member in room.get_joined_members()}
+                except MatrixRequestError as e:
+                    last_ex = e
+                room_is_empty = not bool(peer_ids & member_ids)
+                if room_is_empty or last_ex:
+                    if last_ex:
+                        raise last_ex
```

Comment:
What's the point of raising `last_ex` inside `for retry` loop? It's the same as not handling the exception at all. The `raise last_ex` inside `else` of `for` loop is a patter to allow swallowing exceptions a given number of times, and re-raising them if `for` couldn't succeed (`break`) on the retries. Re-raising inside the loop is the same as not catching it

Reply:
The idea is to only raise the exception once the retries have been exhausted, as the server response might change in the meantime.

Reply:
I know. That's the pattern, but the code there raises on the first exception, which is the same as not handling it.

Reply:
Aaah sorry, didn't see it was there twice